### DISCUSSION TITLE
How to use nextjs13 with custom fastify server?

### DISCUSSION THREAD

**Discussion by BismitaRath** [vercel/next.js#58118] [App Router]
I have a custom fastify server integrated with nextjs. I am updating from next.js version 12 to next.js 13. My usecase is I have some plugins registerd in server side and some request decorater as well. But in app folder this is not accessible as app folder doesnt have access to server request. I am using `@fastify/nextjs` to forward fastify request to nextjs server as well, But is seems nextjs side it is treating as a new request. Any way or example I can follow to solve this issue? My issue is to access this fastify request context in app folder. 
 ```
      fastify.register(require("@fastify/nextjs"), { dev, hostname, port }).after(() => {
          server.next("/api/*", { method: "GET" });
          server.next("/api/*", { method: "POST" });
          server.next("/*", { preHandler }, some-render-logic);
        });
   ```

**Comment by GauravKumarSingh** [comment]
Even i am facing the similar issue.

**Comment by BismitaRath** [comment]
@leerob Please share your insight on this.

**Comment by arunvishnun** [comment]
The documentation for custom server disappeared with app router introduction. This probably could be a common use case, particularly for teams that have adopted Next.js version 12 or earlier and now find themselves unable to proceed further, resulting in less flexibility with their upgrades. 
What is the official guidance on using a custom server going forward?
